### PR TITLE
Fix codestandards 402

### PR DESCRIPTION
**Pull request by matthewhilton**
**Changes**
- A port of https://github.com/catalyst/moodle-tool_objectfs/pull/620 to `MOODLE_402_STABLE` which fixes things with `phpcbf` as well as manually fixing unit test data providers
- Also copied over a commit made to 310 but not 402 - 00b514ffcc8da92ca8f1bb71624b382862a023f0 to fix phpdocs as well

### CODE REVIEW

File: classes/local/object_manipulator/candidates/candidates_factory.php [hunk 1]
```diff
@@ -33,6 +33,9 @@
 use tool_objectfs\local\object_manipulator\recoverer;
 use tool_objectfs\local\object_manipulator\orphaner;
 
+/**
+ * Candiates Factory
```
Comment (danmarsden) [inline classes/local/object_manipulator/candidates/candidates_factory.php]: is this a factory where candy is eaten or just a typo? :-)

File: classes/local/object_manipulator/candidates/candidates_finder.php [hunk 2]
```diff
@@ -27,6 +27,9 @@
 use moodle_exception;
 use stdClass;
 
+/**
+ * Candiates Finder
```
Comment (danmarsden) [inline classes/local/object_manipulator/candidates/candidates_finder.php]: another one.

File: classes/privacy/provider.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ class provider implements null_provider {
      *
      * @return  string
      */
-    public static function _get_reason() {
+    public static function get_reason(): string {
```
Comment (danmarsden) [inline classes/privacy/provider.php:37]: this looks like a conversion from using the old legacy polyfill, but the use legacy_polyfill is still in place?

File: classes/local/object_manipulator/candidates/candidates_factory.php (hunk 1 above)
Comment (matthewhilton) [reply classes/local/object_manipulator/candidates/candidates_factory.php]: Just a typo :)

File: classes/local/object_manipulator/candidates/candidates_finder.php (hunk 2 above)
Comment (matthewhilton) [reply classes/local/object_manipulator/candidates/candidates_finder.php]: Fixed

File: classes/privacy/provider.php (hunk 3 above)
Comment (matthewhilton) [reply classes/privacy/provider.php:37]: Ah yep, good pickup. Wasn't aware of the polyfill, and have removed since its no longer necessary.

**Pull request merged**